File: android/app/src/main/java/org/fossasia/openevent/data/facebook/CommentItem.java
package org.fossasia.openevent.data.facebook;

import android.os.Parcel;
import android.os.Parcelable;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.PropertyNamingStrategy;
import com.fasterxml.jackson.databind.annotation.JsonNaming;

import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Created by rohanagarwal94 on 11/6/17.
 */

@Data
@NoArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
public class CommentItem implements Parcelable {

    private String createdTime;
    private Commenter from;
    private String message;
    private String id;

    protected CommentItem(Parcel in) {
        createdTime = in.readString();
        message = in.readString();
        id = in.readString();
    }

    public static final Creator<CommentItem> CREATOR = new Creator<CommentItem>() {
        @Override
        public CommentItem createFromParcel(Parcel in) {
            return new CommentItem(in);
        }

        @Override
        public CommentItem[] newArray(int size) {
            return new CommentItem[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(createdTime);
        dest.writeString(message);
        dest.writeString(id);
    }
}

File: android/app/src/main/java/org/fossasia/openevent/core/feed/facebook/api/CommentItem.kt
package org.fossasia.openevent.core.feed.facebook.api

import android.os.Parcelable
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.databind.PropertyNamingStrategy
import com.fasterxml.jackson.databind.annotation.JsonNaming
import kotlinx.android.parcel.Parcelize

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy::class)
@Parcelize
data class CommentItem(
        val createdTime: String? = null,
        val from: Commenter? = null,
        val message: String? = null,
        val id: String? = null
): Parcelable